lib/semantics/check-do.h

```diff
@@ -25,6 +25,8 @@ struct ExitStmt;
 
 namespace Fortran::semantics {
 
+using NamePtr = parser::Name const *;
```

Comment:
In my opinion this type alias just obscures things and it would be better to use `const parser::Name *` everywhere.

Reply:
OK.  I'll make that change.